prog/postgres/converge_postgres_resource.rb

```diff
@@ -24,7 +24,10 @@ class Prog::Postgres::ConvergePostgresResource < Prog::Base
 
   label def wait_servers_to_be_ready
     hop_provision_servers unless postgres_resource.has_enough_fresh_servers?
-    hop_recycle_representative_server if postgres_resource.has_enough_ready_servers?
+    if postgres_resource.has_enough_ready_servers?
+      register_deadline(nil, (24 - PostgresResource::MAINTENANCE_DURATION_IN_HOURS) * 60 * 60)
```

Comment:
22 hours of deadline is pretty long. I think instead, we can set a short deadline just after the line we nap for 10 minutes (in next label)

Reply:
Nope, that's it. I was kind of worried that we wouldn't be notified if somehow we messed and some servers are infinitely waiting but that's the case for many other operations in the codebase as well. I have updated it accordingly